### ISSUE TITLE
Update 3, no nvme volume after reboot, no E10M20 functionality

### ISSUE THREAD

**Issue by zcpnate** [007revad/Synology_HDD_db#132]
I shouldn't have done it... I did it 🙃 

Setup:
DS1821+
E10M20-TI with 2x WD Red nvme Operating as R/W Cache for Volume 1
Volume 1: 4x SHR-2 HDD
Volume 2 - Internal nvme slots: 2x 970 Evo Plus SHR-1 Volume 

---

Attempted to Upgrade from Update 1 > Update 3 after having issues on Update 2 and rolling back to Update 1.

Alas, similar or even worse issues on Update 3. Let me recount:

Update 3 applied. No forced reboot - everything operating normally.

Ran script 3.1.62 manually to ensure proper settings. Manually Reboot

hdd_db script scheduled to run at shutdown and boot

After reboot, no nvme volumes appear. Volume 2 missing, HDD Volume 1 Critical no nvme Cache

Drives and E10M20 show in script output but do not have any functionality from this point on. LAN 5 from the E10M20 does appear to show up.

Attempt to force downgrade to update 1 by running the reinstall script and then applying PAT file.

After reboot, this does resolve the issue with Volume 2 it appears and is functional after "update" reboot.

E10M20 still only functional for LAN thus Volume 1 remains critical. 

However, even after applying the PAT, info still shows DS remains on update 3. 

After rebooting once more, back to entirely broken state.

**Comment by zcpnate** [comment]
Update:

I've attempted telnet version downgrade and even the process described here using a spare drive:
https://www.reddit.com/r/synology/comments/140ywsm/how_i_broke_dsm_and_recovered_without_losing_my/

No matter what, after applying the PAT for 7.2U1 and then getting into DSM, we're on Update 3

So I guess the options are:
1) Figure out a way to downgrade to U1 where everything works perfectly.
or 
2) Make E10M20 and NVME volumes work on U3.

Thoughts?

**Comment by zcpnate** [comment]
I've gotten back to Update 1!

After running the reinstall version script and rebooting, we're met with Synology Web Assistant.

I had the thought to block all outbound internet traffic from the DS as I had a hunch it might be auto-applying the Update 2 and Update 3 patches as part of the restore after manually selecting Update 1 PAT.

I was right.

However, the DS does require internet connection to begin the update with local pat. Yikes.

I enabled the firewall policy to allow traffic for a moment, started the upgrade and then blocked all outbound internet traffic.

Success, restore complete on update 1 - data volumes back. A bit of resetup to do since I deleted settings and restored throughout my debugging.

Domains contacted by web assistant that appear relevant:
update7.synology.com
dataupdate7.synology.com
dataautoupdate7.synology.com

So basically I feel like I should never update again lol

**Comment by 007revad** [comment]
> So basically I feel like I should never update again lol

A few other people said this after update 2 was released. I've been too scared to update my DS1821+ to update 2 or 3. Though I did update my DS720+ test NAS without any issues.

> Domains contacted by web assistant that appear relevant: 
> update7.synology.com 
> dataupdate7.synology.com 
> dataautoupdate7.synology.com

Did you capture more than just the domains?

Do you have auto update enabled for DSM updates?

Many synology.com addresses are configurable/editable in 4 different files on the DS1821+ but "dataupdate7.synology.com" is hardcoded in 2 library files. I can't tell if the libraries only use the hard coded URLs if the URL is not configured in synoinfo.conf - like the drive_db_test_url that we can add to synoinfo.conf to prevent DSM updating the drive databases with `drive_db_test_url=127.0.0.1`

And I can't find "dataautoupdate7.synology.com" in any of DSM 7.2 Update 1's files.

**Synoinfo.conf contains:**
pkgupdate_server="https://pkgupdate7.synology.com"
rss_server="http://update7.synology.com/autoupdate/genRSS.php"
rss_server_ssl="https://update7.synology.com/autoupdate/genRSS.php"
rss_server_v2="https://update7.synology.com/autoupdate/v2/getList"
security_version_server="https://update7.synology.com/securityVersion"
small_info_path="https://update7.synology.com/smallupdate"
update_server="http://update7.synology.com/"
update_server_ssl="https://update7.synology.com/"
updateurl="http://www.synology.com/"


**Files containing "dataupdate7.synology.com"**

**`https://dataupdate7.synology.com/` is hardcoded in libhwcontrol.so and libsynoinstallcpp.so**
- libhwcontrol.so seems to only use `https://dataupdate7.synology.com/` to get Synology drive firmware updates.
- libsynoinstallcpp.so seems to only use `https://dataupdate7.synology.com/` to get DSM small (nano) updates via `https://dataupdate7.synology.com/smallupdates`.

**/usr/share/data_update/data_update.conf:**
```
{
	"config": {
		"download_scheme": "https",
		"db_netloc": "dataupdate7.synology.com",
		"db_ver": "v1",
		"cur_volume": ""
	}
}

```
**/usr/syno.sbin/junior_expansion_pack.sh:**
```
get_update_server_hostname() {
	echo "dataupdate7.synology.com"
}
```

**/usr/syno/bin/securityscan/Define.py:**
```
# Demain name and URL for DB info
DB_URL_DOMAIN_NAME          = "dataupdate7.synology.com"
```

**Comment by BackedUpBooty** [comment]
is there any more word on whether this will work with update 3? the work you've done is top notch, but I'm like you with your 1821+, not brave enough to attempt update past 7.2U1

**Comment by 007revad** [comment]
I updated my DS1821+ a week ago to update 3 and haven't had any issues.

My DS720+ updated itself(!) to update 3 when I ran my [Synology_DSM_reinstall](https://github.com/007revad/Synology_DSM_reinstall) script to reinstall DSM 7.2 with update 1.  I have it set to not auto update.

**Comment by BackedUpBooty** [comment]
Well that's good news. I would have thought more people would have posted if there were wider-ranging issues with U3 as well, so I may bite the bullet myself over the weekend.

**Comment by 007revad** [comment]
Agreed. I only know of a very small number of people who had issues after updating to update 2 or update 3.

**Comment by 007revad** [comment]
> And I can't find "dataautoupdate7.synology.com" in any of DSM 7.2 Update 1's files.

To answer my own question:

**Synology HDD/SSD Update Pack Download Site**
- dataupdate7.synology.com
- dataautoupdate7.synology.com
- dataautoupdate7.synology.cn (Only for China)

**DSM Update Download Site**
- global.download.synology.com
- global.synologydownload.com
- update7.synology.com
- autoupdate7.synology.com
- autoupdate7.synology.cn (Only for China)

Source: https://kb.synology.com/en-my/DSM/tutorial/What_websites_does_Synology_NAS_connect_to_when_running_services_or_updating_software

**Comment by riverflow** [comment]
I recently bought a ds1821. I updated dsm to UPDATE 3 and installed packages. I wanted to use shr-6 ssd cache, so I bought E10M20, and ran script. but like zcpnate, all ssd cache was gone. so I restored script..

are there no option for use E10M20 on UPDATE 3?

how about device name change on script?  ex. E10M20-T1 to E10G18-T2

**Comment by 007revad** [comment]
Do you want the NVMe drives in the E10M20-T1 as cache drives or a volume?

Can you try https://github.com/007revad/Synology_enable_M2_card

Looks like I might need to buy a E10M20-T1 and do some testing.

**Comment by BackedUpBooty** [comment]
Just did the update, can confirm no issues with the Samsung NVMe sticks as volume in my 920+. Docker spun right back up and all is good.

**Comment by riverflow** [comment]
> Do you want the NVMe drives in the E10M20-T1 as cache drives or a volume?
> 
> Can you try https://github.com/007revad/Synology_enable_M2_card
> 
> Looks like I might need to buy a E10M20-T1 and do some testing.

I ran Synology_enable_M2_card.sh on my DS1821+ dsm7.2U3 with E10M20-T1,  same phenomenon.
All M2 drives were gone.
I did restoration.

**Comment by 007revad** [comment]
There's a new release candidate version of the script that now correctly enables M.2 cards like the E10M20-T1 for the DS1821+ and DS1621+

https://github.com/007revad/Synology_HDD_db/releases/tag/v3.2.67-RC

**Comment by 007revad** [comment]
[Synology_HDD_db v3.2.68](https://github.com/007revad/Synology_HDD_db/releases/tag/v3.2.68) released which now correctly, and simply, enables E10M20-T1, M2D20, M2D18 and M2D17 in models that use device tree and are using DSM 7.2 Update 2 and 3, 7.2.1, 7.2.1 Update 1, 2 and 3.

I also updated [Synology_enable_M2_card](https://github.com/007revad/Synology_enable_M2_card) which does the same but allows you choose which M.2 card to enable, or choose to enable all of them.
